### ISSUE TITLE
Wanted: Support Native ESM

### ISSUE THREAD

**Issue by zhmushan** [tusen-ai/naive-ui#3651]
### This function solves the problem (这个功能解决的问题)
NaiveUI ESM doesn't work on Nodejs and Browser.

I want to import NaiveUI on native Nodejs to help me generate a static site, but the output of NaiveUI ESM does not work directly, all the import and export paths are missing the `.js` suffix, and the Nodejs ESM resolver does not automatically add the suffix name.
### Expected API (期望的 API)
I don't know anything about the bundling tools used by NaiveUI.

I checked the ESM output directly, and if we want to support Nodejs, we need to find all statements that import and export modules via relative paths but omit the file suffixes, and complete the paths in all of them. If we also want to support Browser, we need to find all the modules in the output whose paths are resolved to `node_modules`, inline them, or generate `importmap`.

**Comment by 07akioni** [comment]
It would be fucking complex to add such suffix.

**Comment by zhmushan** [comment]
There is a simple and stupid way: just add the `.js` suffix to the import/export statements of all source files and there will be no need to add additional steps to the bundling process. (Inspired by [Astro](https://github.com/withastro/astro))

**Comment by 07akioni** [comment]
> st add the `.js` suffix to the import/export statements of all source files and there will be no need to add additional steps to the bundling proces

No, it's not possible unless it's created as pure ESM module, which is impossible now.
